cmd/node/factory/interface.go

```diff
@@ -16,6 +16,7 @@ type HeaderSigVerifierHandler interface {
 	VerifySignature(header data.HeaderHandler) error
 	VerifySignatureForHash(header data.HeaderHandler, hash []byte, pubkeysBitmap []byte, signature []byte) error
 	VerifyPreviousBlockProof(header data.HeaderHandler) error
```

Comment:
do we still need the VerifyPreviousBlockProof? it seems a bit redundant as it would simply extract the proof from the header and call VerifyHeaderProof on it right? Does it also verify the hash is previousBlockHash?